hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/handlers/FileAppendHandler.java

```diff
@@ -140,6 +143,42 @@ public void handle(@NonNull final HandleContext handleContext) throws HandleExce
         fileStore.put(fileBuilder.build());
     }
 
+    private void feeCalculation(
+            HandleContext handleContext,
+            FileAppendTransactionBody fileAppend,
+            File file,
+            @NonNull FilesConfig fileServiceConfig) {
+        final var dataLength =
+                (fileAppend.contents() != null) ? fileAppend.contents().length() : 0;
+
+        /**
+         * TODO: revisit after modularizaion completed
+         * PR conversation: 8089
+         */
+        final long effectiveLifeTime;
+        final var fileNum = file.fileId().fileNum();
+        final var firstSoftwareUpdateFile =
+                fileServiceConfig.softwareUpdateRange().left();
+        final var lastSoftwareUpdateFile =
+                fileServiceConfig.softwareUpdateRange().right();
+        if (firstSoftwareUpdateFile <= fileNum && fileNum <= lastSoftwareUpdateFile) {
+            effectiveLifeTime = THREE_MONTHS_IN_SECONDS;
+        } else {
+            final var effCreationTime =
+                    handleContext.body().transactionID().transactionValidStart().seconds();
+            final var effExpiration = (file.expirationSecond() > 0) ? file.expirationSecond() : effCreationTime;
+            effectiveLifeTime = effExpiration - effCreationTime;
+        }
+
+        final var fees = handleContext
+                .feeCalculator(SubType.DEFAULT)
+                .addBytesPerTransaction(BASIC_ENTITY_ID_SIZE + dataLength)
+                .addStorageBytesSeconds(dataLength * effectiveLifeTime)
+                .calculate();
+
+        handleContext.feeAccumulator().charge(handleContext.payer(), fees);
```

Comment:
This has to be done differently now. The initial example was wrong. Just a heads up, should probably be done in a separate PR.

Reply:
can you attach a right example and I will open a ticket for it.